lib/rules/forbid-component-props.js

```diff
@@ -22,48 +31,117 @@ module.exports = {
       recommended: false
     },
 
-    schema: [{
-      type: 'object',
-      properties: {
-        forbid: {
-          type: 'array',
-          items: {
-            type: 'string'
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          defaultAllowOnDOM: {
+            type: 'boolean'
+          },
+          forbid: {
+            type: 'array',
+            items: {
+              oneOf: [
+                {
+                  type: 'string'
+                },
+                {
+                  type: 'object',
+                  properties: {
+                    property: {
+                      type: 'string'
+                    },
+                    allowOnDOM: {
+                      type: 'boolean'
+                    },
+                    allowOnComponents: {
+                      type: 'boolean'
+                    },
+                    forbid: {
+                      type: 'array',
+                      items: {
+                        type: 'string'
+                      }
+                    }
+                  }
+                }
+              ]
+            }
           }
-        }
-      },
-      additionalProperties: true
-    }]
+        },
+        additionalProperties: true
```

Comment:
i think we can maybe set this to `false`, so nobody accidentally passes an invalid config option?